apps/api/src/rating/rating.service.ts

```diff
@@ -42,11 +44,10 @@ export class RatingService {
     req: any,
     createRatingDto: CreateRatingDto,
   ): Promise<Rating> {
-    const userId = req._id;
-    const authUser = await this.authService.findOne(userId);
+    const userId = await this.userService.findUser(req);
     let itemFound = false;
     const allDeliveredOrders = await this.orderService.getAllDeliveredOrders(
-      userId,
+      req,
     );
 
     for (const order of allDeliveredOrders) {
```

Comment:
this will remove the flag and function the same way

```suggestion
    const itemToRate = allDeliveredOrders.find((order) =>
      order.orderedItems.find(
        (item) => item.productId.toString() === createRatingDto.productId,
      ),
    );

    if (itemToRate) {
      const existingRating = await this.ratingModel.findOne({
        userId,
        productId: createRatingDto.productId,
      });

      if (existingRating) {
        throw new BadRequestException(
          'Already provided rating for this product!',
        );
      }
      const now = new Date();
      const rating = {
        ...createRatingDto,
        userId,
        username: userId.auth.username,
        publishedDate: now,
        reviewDate: now,
      };

      const userRating = await this.ratingModel.create(rating);
      return userRating;
    } else {
      throw new NotFoundException('Product not found in delivered order');
    }
```

Reply:
done, monsieur